include/seqan3/range/container/concatenated_sequences.hpp

```diff
@@ -181,7 +181,35 @@ class concatenated_sequences
     using size_type = ranges::size_type_t<data_delimiters_type>;
     //!\}
 
-    /* rule of six */
+protected:
+    //!\cond
+    // unfortunately we cannot specialise the variable template so we have to add an auxiliary here
+    template <typename t>
+        requires (dimension_v<t> == dimension_v<value_type> + 1) &&
+                 std::is_same_v<remove_cvref_t<innermost_value_type_t<value_type>>,
+                                remove_cvref_t<innermost_value_type_t<t>>>
+    static constexpr bool is_compatible_this_aux = true;
+    //!\endcond
+
+    //!\brief Whether a type satisfies seqan3::compatible_concept with this class.
+    //!\hideinitializer
+    // cannot use the concept, because this class is not yet fully defined
+    template <typename t>
+    static constexpr bool is_compatible_this = is_compatible_this_aux<t>                                    ||
+                                               std::is_same_v<remove_cvref_t<t>, concatenated_sequences>    ||
+                                               std::is_same_v<remove_cvref_t<t>, iterator>                  ||
+                                               std::is_same_v<remove_cvref_t<t>, const_iterator>;
+
+    //!\brief Whether a type satisfies seqan3::compatible_concept with this class's value_type or reference type.
+    //!\hideinitializer
+    // we explicitly check same-ness, because these types may not be fully resolved, yet
+    template <typename t>
+    static constexpr bool is_compatible_value = compatible_concept<value_type, t>                   ||
+                                                std::is_same_v<remove_cvref_t<t>, value_type>       ||
+                                                std::is_same_v<remove_cvref_t<t>, reference>        ||
+                                                std::is_same_v<remove_cvref_t<t>, const_reference>;
```

Comment:
Oh man, really the value_type might not be completely there? Can you give an example?

Reply:
as discussed :skull: